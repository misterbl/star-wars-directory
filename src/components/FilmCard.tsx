import React from "react";
import charactersPhotos from "../charactersPhotos";
import { withRouter } from "react-router";
import { bindActionCreators, Dispatch } from "redux";
import { connect } from "react-redux";
import singleSearch from "../actions/thunks/singleSearch";
import { IFilmCardComponent } from "./FilmCard.d";
import ROUTES from "../routes";
import formattedDate from "../utils/formattedDate";

export const FilmCard = (props: IFilmCardComponent) => {
  const {
    film,
    history: { push }
  } = props;
  const onClick = () => {
    props.singleSearch(film.url);
    push(ROUTES.FILM);
  };

  return (
    <div onClick={onClick} className="pointer m-3 text-white">
      <u className="card-text">{film.title.toUpperCase()}</u>
      <div className="d-flex">
        <div className="avatar avatar__default card-img-top" />
        <div className="ml-3">
          <p>{`Director: ${film.director}`}</p>
          <p>{`Released the ${formattedDate(film.release_date)}`}</p>
        </div>
      </div>
    </div>
  );
};

export const mapDispatchToProps = (dispatch: Dispatch) => ({
  singleSearch: bindActionCreators(singleSearch, dispatch)
});

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(FilmCard)
);
