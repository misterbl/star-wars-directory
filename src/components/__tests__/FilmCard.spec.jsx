/* eslint-disable no-undef */
import * as React from "react";
import { shallow } from "enzyme";
import history from "../../testMocks/history.mock";
import { FilmCard } from "../FilmCard";
import ROUTES from "../../routes";

describe("FilmCard", () => {
  const props = {
    history,
    singleSearch: jest.fn(),
    film: {
      director: "name",
      release_date: "1222",
      title: "title"
    }
  };
  const wrapper = shallow(<FilmCard {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls singleSearch() onClick ", () => {
    const mainDiv = wrapper.find("div").at(0);
    mainDiv.simulate("click");
    expect(props.singleSearch).toHaveBeenCalled();
  });
  it("calls push() onClick ", () => {
    const mainDiv = wrapper.find("div").at(0);
    mainDiv.simulate("click");
    expect(props.history.push).toHaveBeenCalledWith(ROUTES.FILM);
  });
});
