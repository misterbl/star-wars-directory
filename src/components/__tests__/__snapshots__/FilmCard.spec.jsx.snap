// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilmCard
    assignCurrentView={[MockFunction]}
    film={
      Object {
        "director": "name",
        "release_date": "1222",
        "title": "title",
      }
    }
    history={
      Object {
        "action": "PUSH",
        "block": [MockFunction],
        "createHref": [MockFunction],
        "go": [MockFunction],
        "goBack": [MockFunction],
        "goForward": [MockFunction],
        "length": 123,
        "listen": [MockFunction],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": "",
        },
        "push": [MockFunction],
        "replace": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <u
          className="card-text"
        >
          TITLE
        </u>,
        <div
          className="d-flex"
        >
          <div
            className="avatar avatar__film"
          />
          <div
            className="ml-3 p-3"
          >
            <p>
              Director: name
            </p>
            <p>
              Released the 31th December 1221
            </p>
          </div>
        </div>,
      ],
      "className": "pointer film-card m-3 text-white border border-dark p-3",
      "onClick": [Function],
      "role": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TITLE",
          "className": "card-text",
        },
        "ref": null,
        "rendered": "TITLE",
        "type": "u",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="avatar avatar__film"
            />,
            <div
              className="ml-3 p-3"
            >
              <p>
                Director: name
              </p>
              <p>
                Released the 31th December 1221
              </p>
            </div>,
          ],
          "className": "d-flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "avatar avatar__film",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Director: name
                </p>,
                <p>
                  Released the 31th December 1221
                </p>,
              ],
              "className": "ml-3 p-3",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Director: name",
                },
                "ref": null,
                "rendered": "Director: name",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Released the 31th December 1221",
                },
                "ref": null,
                "rendered": "Released the 31th December 1221",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <u
            className="card-text"
          >
            TITLE
          </u>,
          <div
            className="d-flex"
          >
            <div
              className="avatar avatar__film"
            />
            <div
              className="ml-3 p-3"
            >
              <p>
                Director: name
              </p>
              <p>
                Released the 31th December 1221
              </p>
            </div>
          </div>,
        ],
        "className": "pointer film-card m-3 text-white border border-dark p-3",
        "onClick": [Function],
        "role": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TITLE",
            "className": "card-text",
          },
          "ref": null,
          "rendered": "TITLE",
          "type": "u",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="avatar avatar__film"
              />,
              <div
                className="ml-3 p-3"
              >
                <p>
                  Director: name
                </p>
                <p>
                  Released the 31th December 1221
                </p>
              </div>,
            ],
            "className": "d-flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "avatar avatar__film",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Director: name
                  </p>,
                  <p>
                    Released the 31th December 1221
                  </p>,
                ],
                "className": "ml-3 p-3",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Director: name",
                  },
                  "ref": null,
                  "rendered": "Director: name",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Released the 31th December 1221",
                  },
                  "ref": null,
                  "rendered": "Released the 31th December 1221",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
